dev-tool-requirements.txt

```diff
@@ -0,0 +1,2 @@
+nox==2023.4.22
+poetry==1.5.1
```

Comment:
do we want to use this very specific versions?

Reply:
It could prevent updates to Nox or Poetry causing our builds to break, which might be hard to and annoying to debug. I think the best practice would be to leave it like this and let dependabot suggest updates... Maybe I need to make changes to the dependabot.yml, forgot to do that

Reply:
Updated!